hazelcast/src/test/java/com/hazelcast/internal/util/ServiceLoaderTest.java

```diff
@@ -101,6 +101,42 @@ public void testMultipleClassloaderLoadsTheSameClass() throws Exception {
         }
     }
 
+    @Test
+    public void testMultipleClassloaderLoadsTheSameClass_fromParentClassLoader() throws Exception {
```

Comment:
any idea how a proper integration test could look like?

Reply:
Added one [here](https://github.com/hazelcast/hazelcast/pull/18103/commits/e6d3ed5ca35b5250e242f60a5244de0b47ddaf67), as a quick test to verify it works on jenkins PR builder.